### DISCUSSION TITLE
Authorization for component

### DISCUSSION THREAD

**Discussion by davideperfetto** [livewire/livewire#1948] [1. Help] [answered]
Quick question: suppose i set a middleware on a full page component(ex auth), can I safely assume that will not be possible to call an action of that component if not logged in?

Similarly, if I call the authorize function on the mount method of a component, can I assume that will not be possible to call any action of the component if not authorized?

I am asking this because I am trying to figure out if is it possible to just set a guard via middleware or just on the mount method instead of authorizing any single action of the component

**Comment by spaantje** [comment]
Hi @davideperfetto did you ever find an answer to this question? I did find [this article](https://www.matthewerwin.co.uk/blog/protecting-livewire-components-when-using-multiple-auth-drivers/) which recommends adding the authorization to the `__construct()`. But i don't really know if it's necessary if the component is wrapped with authorization on the page where it's included?

**Reply by joshhanley** [reply]
@spaantje just an FYI, the constructor is not considered part of the public API for Livewire, so you shouldn't use that at all. Use mount instead.

To answer the question, I think it would be ok to add authorisation to the route or even in the mount method and it should be protected, as the Livewire endpoint requires an already signed request with a checksum that can only be generated from inside the component when it loads on the first request.

But, I think you can never be too careful, so if you want to ensure all subsequent requests are also protected, you could add authorisation to the `hydrate` hook method.

This means authorisation in mount will run on the first request and authorisation in hydrate will run on any subsequent requests.

That way you are fully covered 🙂

Hope this helps!

**Reply by spaantje** [reply]
Thanks!

**Reply by davideperfetto** [reply]
Comfirm what @joshhanley said.

I have to edit my previous response. Putting the auth logic in the hydrate method is correct. However it is not necessary thanks to the persistent middleware logic
https://laravel-livewire.com/docs/2.x/security


> The second security measure Livewire puts in place is "persistent middleware". This means Livewire will capture any authentication/authorization middleware that was used during the "Initial Request" and re-apply it to subsequent requests

**Reply by davideperfetto** [reply]
@spaantje I read  the [article you linked](https://www.matthewerwin.co.uk/blog/protecting-livewire-components-when-using-multiple-auth-drivers/) again. What it suggest is to create a trait and create the auth check in the inizialize method of the trait.

This feature does not seems to be described in [traits section](https://laravel-livewire.com/docs/2.x/traits) of the livewire docs, however upon initialization, livewire call any initializeTraitName of the traits used in a class.

[](https://github.com/livewire/livewire/blob/6bae6be4095155776b78025e0425e6f3cf7f572f/src/Component.php)

@calebporzio Can you add it in the docs?

Thanks

**Reply by joshhanley** [reply]
@davideperfetto its undocumented as it was considered an internal API. But I think it's something that should probably be documented as plenty of people use it. Documentation PRs are welcome at https://github.com/livewire/docs

**Reply by davideperfetto** [reply]
@joshhanley  https://github.com/livewire/docs/pull/376

**Comment by spaantje** [comment]
@davideperfetto The article actually suggest 2 things. The first one is adding it to the `constructor`. And the second one is creating a `trait` so it can be reused. But because i couldn't find any other recommendations about this and also your question here i asked. 

I think i use the answer for @joshhanley from now on. (Adding it to the `mount` method; and maybe to the `hydrate` just to be sure.)

Thanks to you both for the replies!

**Comment by spaantje** [comment, marked as answer]
It looks like the `boot()` method is the best place to do it from version 2.6 onwards.

Based on this PR: https://github.com/livewire/docs/pull/376 & the lifecycle docs: https://laravel-livewire.com/docs/2.x/lifecycle-hooks

(Just updating this for when someone finds this question).
